lib/vintage_net_ethernet.ex

```diff
@@ -38,10 +44,31 @@ defmodule VintageNetEthernet do
   @impl VintageNet.Technology
   def normalize(%{type: __MODULE__} = config) do
     config
+    |> normalize_mac_address()
     |> IPv4Config.normalize()
     |> DhcpdConfig.normalize()
   end
 
+  defp normalize_mac_address(%{mac_address: mac_address} = config) do
+    if not valid_mac?(mac_address) do
+      raise ArgumentError, "Invalid MAC address #{inspect(mac_address)}"
+    end
+
+    config
+  end
+
+  defp normalize_mac_address(config), do: config
+
+  defp valid_mac?(mac_address) when is_binary(mac_address) do
```

Comment:
```suggestion
  defp valid_mac?(<<_::binary-2, ":", _::binary-2, ":", _::binary-2, ":", _::binary-2, ":", _::binary-2, ":", _::binary-2) do
```
What if we did a really explicit match here to ensure it's correctly formatted so it's actually considered valid instead of any random 17 bytes.

We could even check each 2 byte pair to ensure they are hex or numeric as well

Reply:
Good idea. I just pushed a MAC address utility module with the additional checks. 

Before you ask, I expect to eventually move this to `vintage_net`. This whole feature can be applied to other technologies and probably will be needed for WiFi at some point anyway.